source/client.c

```diff
@@ -1788,3 +1779,150 @@ int aws_mqtt_client_connection_ping(struct aws_mqtt_client_connection *connectio
 
     return AWS_OP_SUCCESS;
 }
+
+/*******************************************************************************
+ * Resubscribe
+ ******************************************************************************/
+
+/* The lifetime of this struct is from subscribe -> suback */
+struct resubscribe_task_arg {
+
+    struct aws_mqtt_client_connection *connection;
+
+    /* list of subscribe_task_topic *s */
+    struct aws_array_list topics;
+
+    /* Packet to populate */
+    struct aws_mqtt_packet_subscribe subscribe;
+};
+
+static bool s_reconnect_resub_iterator(const struct aws_byte_cursor *topic, enum aws_mqtt_qos qos, void *user_data) {
+    struct resubscribe_task_arg *task_arg = user_data;
+
+    struct aws_mqtt_topic_subscription sub;
+    AWS_ZERO_STRUCT(sub);
+    sub.topic = *topic;
+    sub.qos = qos;
+
+    aws_array_list_push_back(&task_arg->topics, &sub);
```

Comment:
What happens if the user's original subscribe hasn't gotten its suback callback yet? Is that subscribe command in a queue somewhere? Will issuing this giant resubscribe command somehow interfere with them receiving their suback calback? Are there complications that might arise when we haven't sent the initial SUBSCRIBE vs when we lost the connection before the SUBACK came back?

just peppering you with every scenario I can imagine